src/main/java/com/aws/greengrass/clientdevices/auth/api/UseCases.java

```diff
@@ -12,7 +12,15 @@ public class UseCases {
     private Context context;
 
     public interface UseCase<R, D>  {
-        Result<R> apply(D var1);
+        @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+        Result<R> apply(D var1) throws Exception;
+    }
+
+    public UseCases() {
+    }
```

Comment:
do we need this one?

Reply:
Yes we need it, if we add another constructor like the one below and do no add this one then it can't be auto created by the `Context` and a whole bunch of tests fail